stable/logdna-agent/values.yaml

```diff
@@ -0,0 +1,22 @@
+image:
+  name: logdna/logdna-agent
+  tag: latest
+  pullPolicy: Always
```

Comment:
Fix a release version and `IfNotPresent`

Reply:
As a part of our product differentiation, we do employ an auto-update mechanism to our agents (both Linux native and Kubernetes). This `pullPolicy` is required in order for that mechanism to work with Kubernetes. Our official daemonset has this as well: https://github.com/logdna/logdna-agent/blob/master/logdna-agent-ds.yaml. We're hoping we can get an exception here.

Reply:
This is not ideal. It is general best practice to not use latest versions in a production environment. You offer fixed releases on Docker Hub which you should use. Add `appVersion` in `Chart.yaml` which should match that.

Reply:
Hmmm this would create serious issues with our existing autoupdate mechanism.

Can I propose this? If this is acceptable, we'll build this out.
- By default, it will use latest and pull policy always (this allows us to maintain autoupdate as a key feature)
- We'll allow a values override either in the yaml or command line which will instruct the agent to avoid autoupdate for this particular deployment...this'll give users the flexibility to lock it to a particular version that we offer in Docker Hub

Reply:
Releases are supposed to be stable. You don't want to run with a latest version in production. This is documented in our review guidelines.

https://github.com/kubernetes/charts/blob/master/REVIEW_GUIDELINES.md#configuration

Reply:
Right, I agree that would be a behavior we will allow if that's the desire of the person installing the chart (by specifying values). I'd just rather not have it be the default.

We always extensively test our docker images and only build new images relevant for Kubernetes. So each of the docker image versions are essentially more stable versions of our main agent repo for linux (where the image builds from).

Please let us continue to enable autoupdates for the default install. I will include options to allow specific locked versions if needed/desired.

Reply:
@leeliu unfortunately we require a tagged image based on the guidelines @unguiculus linked to. For what it's worth, a latest image tag doesn't actually enable autoupdates as it requires the pod to be restated for the image to be pulled.

Reply:
Ahh, yes. We usually mark old versions for remote upgrade. We detect the running version on our end and remotely restart the container (which triggers the pullPolicy).

The main issue is that the chart will now behave differently than what we normally do, so we actually need to create a specific version of our Kubernetes agent just for Helm that we need to maintain separately from our regular agent (if we kept only 1 version of the agent, we'd end up repeatedly remote killing the container from our end assuming it'll auto pull and it won't, not something we'd want to do).

I understand the need in your guidelines to require tagged images. If I may, I think the intention of that was for something maybe like redis or mongodb or elasticsearch or other software packages where the product is what's being installed where version updates need to be tightly controlled. Our product is mainly a conduit to our cloud solution. Many charts that are currently live now simply use `latest` as a part of their spec because they are agents too, like ours.

I know you have to follow your review guidelines but I'm hoping you can be flexible too on a case-by-case basis. I'm hoping what I proposed above allow the best of both worlds. If someone wants to have a locked version, we'll provide an option. But by default (as @apenney wrote below), we're hoping to continue to maintain a single version for Helm and regular Kubernetes.

If it's really not possible, I still stop asking from here on out but I'm still holding out hope that you will reconsider my proposal.

Reply:
You shouldn't necessarily need to change your process to accommodate charts - you would still be free to use `latest` in your own k8s deploys if you wish (with Helm, using `--set image.tag=latest` or do this via values files). But even if your application is an agent, if it does anything at all, to allow a k8s pod any stability (if a container needs to be recreated for whatever reason) it would need to be pinned to a version. This best practice for k8s is why Charts repo requires this.

Without knowing your application in detail yet, IMO all you would need to do is have *one* or more versioned releases that you support – it should not need to dominate your workflow 😄

Reply:
@scottrigby @prydonius @unguiculus,

This part is also done!